tests/versioning_test.go

```diff
@@ -397,6 +397,82 @@ func (s *versioningIntegSuite) dispatchNewWorkflowStartWorkerFirst() {
 	s.Equal("done!", out)
 }
 
+func (s *versioningIntegSuite) TestDisableLoadUserDataDefaultTasksBecomeUnversioned() {
```

Comment:
it would be nice to have a test for the situation this addresses even with user data (set v1 as default, spool task 1 with build id on v1, spool "use default" task 2, set v2 as default, start workers just for v2 but not v1, test that task 2 gets dispatched to v2). I can work on it after I get back

Reply:
Agree, would you mind adding later?